photoeditor/src/main/java/ja/burhanrashid52/photoeditor/PhotoSaverDirect.java

```diff
@@ -0,0 +1,155 @@
+package ja.burhanrashid52.photoeditor;
+
+import android.graphics.Bitmap;
+import android.graphics.Canvas;
+import android.util.Log;
+import android.view.View;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+
+/**
+ * Created by Burhanuddin Rashid on 19/07/21.
+ *
+ * @author <https://github.com/burhanrashid52>
+ */
+class PhotoSaverDirect {
```

Comment:
I'm not convinced the suffix `Direct` is clear enough, what about `Sync`? The difference between `PhotoSaverDirect` and `PhotoSaverTask` is the synchronous/asynchronous pattern we use to save the image, so the name should clearly reflect that.

Reply:
Believe me, I wish the "Task" method name would be "Async" so I could give "Sync" to the other other.

But as it was "Task", I has to find another name and that was the first came that came to my mind. If the owner wishes he can just rename it, I really don't care :-)

Reply:
I agree with the Sync/Async that should be. I do not care much either, but it would be cleaner (to my view).

Reply:
> 
> 
> I agree with the Sync/Async that should be. I do not care much either, but it would be cleaner (to my view).

Agreed here, too. It's just up to @burhanrashid52 to decide whether he wants to simply rename them. As it is super-easy to do, whenever he wants to do this, he can do himself without any effort.

Reply:
It's a duplicate code of `PhotoEditorTask`. I think it's a write time we switch to `Executors`. Discussed here. https://github.com/burhanrashid52/PhotoEditor/pull/352#discussion_r637492269